jujugui/static/gui/src/app/components/deployment-flow/direct-deploy/direct-deploy.js

```diff
@@ -39,48 +39,80 @@ YUI.add('deployment-direct-deploy', function() {
     },
 
     componentDidMount: function() {
-      if (this.state.isBundle) {
-        this.props.getEntity(this.props.ddData.id, (error, data) => {
-          if (error) {
-            console.error('cannot fetch the entity:' + error);
-            return;
-          }
-          if (data.length > 0) {
-            data = data[0];
-            this.setState({
-              entityModel: this.props.makeEntityModel(data)
-            });
-          }
-        });
-      }
+      this.props.getEntity(this.props.ddData.id, (error, data) => {
+        if (error) {
+          console.error('cannot fetch the entity:' + error);
+          return;
+        }
+        if (data.length > 0) {
+          data = data[0];
+          this.setState({
+            entityModel: this.props.makeEntityModel(data)
+          });
+        }
+      });
     },
 
     render: function() {
       let diagram = null;
-      let description = null;
+      let titleAndDescription = null;
       const ddEntityId = this.props.ddData.id;
       if (this.state.isBundle) {
-        diagram = <juju.components.EntityContentDiagram
-          getDiagramURL={this.props.getDiagramURL}
-          id={ddEntityId} />;
-
-        if (this.state.entityModel) {
-          description = <juju.components.EntityContentDescription
-            entityModel={this.state.entityModel}
-            renderMarkdown={this.props.renderMarkdown}
-            />;
+        diagram = (<div className="six-col deployment-direct-deploy__image">
+            <juju.components.EntityContentDiagram
+              getDiagramURL={this.props.getDiagramURL}
+              id={ddEntityId} />
+          </div>);
+      }
+
+      if (this.state.entityModel) {
+        const entity = this.state.entityModel.toEntity();
+        const description = <juju.components.EntityContentDescription
+          entityModel={this.state.entityModel}
+          renderMarkdown={this.props.renderMarkdown} />;
+        const title = (<h3
+          className="deployment-direct-deploy__title">
+            {entity.displayName}
+          </h3>);
+
+        let url;
+        let link;
+        try {
+          url = window.jujulib.URL.fromLegacyString(ddEntityId);
+        } catch(_) {
+          url = window.jujulib.URL.fromString(ddEntityId);
         }
+        if (url) {
+          link = (<a href={`/${url.path()}`}
```

Comment:
do we need to use `state.generatePath` or whatever it is here to generate this URL correctly if the gui is not deployed under "/"?